tests/output_guardrails/test_generate.py

```diff
@@ -0,0 +1,104 @@
+import json
+from unittest.mock import AsyncMock
+
+import pytest
+
+from govuk_chat_evaluation.output_guardrails.generate import (
+    generate_inputs_to_evaluation_results,
+    generate_and_write_dataset,
+    GenerateInput,
+    EvaluationResult,
+)
+
+
+@pytest.fixture
+def run_rake_task_mock(mocker):
+    async def default_side_effect(_, env):
+        if env["INPUT"] == "Question 1":
+            return {"triggered": True, "llm_guardrail_result": 'True | "1, 4, 7"'}
+        else:
+            return {"triggered": False, "llm_guardrail_result": "False | None"}
+
+    mock = mocker.patch(
+        "govuk_chat_evaluation.output_guardrails.generate.run_rake_task",
+        new_callable=AsyncMock,
+    )
+    mock.side_effect = default_side_effect
+    return mock
+
+
+def test_generate_models_to_evaluation_models_returns_evaluation_results(
+    run_rake_task_mock,
+):
+    def result_per_question(_, env):
+        if env["INPUT"] == "Question 1":
+            return {"triggered": True, "llm_guardrail_result": 'True | "1, 4, 7"'}
+        else:
+            return {"triggered": False, "llm_guardrail_result": "False | None"}
+
+    run_rake_task_mock.side_effect = result_per_question
```

Comment:
I don't think you need this because a side effect is already configured in the mock?